docs/en/operations/external-authenticators/ssl-x509.md

```diff
@@ -0,0 +1,24 @@
+# SSL X.509 certificate authentication {#ssl-external-authentication}
+
+[SSL 'strict' option](../server-configuration-parameters/settings.md#server_configuration_parameters-openssl) enables mandatory certificate validation for the incoming connections. In this case, only connections with trusted certificates can be established. Connections with untrusted certificates will be rejected. Thus, certificate validation allows to uniquely authenticate an incoming connection. `Common Name` field of the certificate is used to identify connected user. This allows to associate multiple certificates with the same user. Additionally, reissuing and revoking of the certificates does not affect the ClickHouse configuration.
+
+To enable SSL certificate authentication, a list of `Common Name`'s for each ClickHouse user must be sspecified in the settings file `config.xml `:
```

Comment:
Is it possible to make SSL certificate authentication optional? For now it sounds like that with 509 client certs the usual SSL connections will not work? Also it's not clear from the doc what will happen with when user have a valid certificate but has unknown common_name. 

So we can have the following situations:

* Client doesn't use the certificate
* Client use some certificate which is invalid
* Client use some certificate that is valid but common_name is not configured
* Client use some certificate which is valid and common_name is not configured

Reply:
SSL certificate authentication is optional. The behavior isn't changed if this option is not activated and in your case it only depends on SSL verificationMode setting. If verificationMode==strict, connections with invalid or missing certificates will be rejected and common_name doesn't matter.

Certificate authentication makes sense only when verificationMode==strict. In this case the above SSL logic holds _and_ ClickHouse checks if proper common_name is set in the certificate. As a result, connection is rejected in case (3) and accepted in case (4)

Reply:
> SSL certificate authentication is optional.  

with `strict` it is NOT optional.  All connections from users w/o certs will be immediately terminated. See https://docs.pocoproject.org/current/Poco.Net.Context.html#28441 

If you need to accept connections both with certificates (in that case we can skip usual authentication) and without that (in that case usual authentication with password is required) you should use `relaxed` not `strict`. 

BTW: maybe having some counters for monitoring rejected SSL connections would be nice.